pkg/scheduler/framework/plugins/podtopologyspread/filtering.go

```diff
@@ -262,6 +264,61 @@ func (pl *PodTopologySpread) calPreFilterState(ctx context.Context, pod *v1.Pod)
 		return &preFilterState{}, nil
 	}
 
+	allPvcs, err := pl.pvcs.PersistentVolumeClaims(pod.Namespace).List(labels.Everything())
+	if err != nil {
+		return nil, fmt.Errorf("listing PersistentVolumeClaims: %w", err)
+	}
+	podNotExist := map[string][]*framework.PodInfo{}
+	processPVC := func(i int) {
+		pvc := allPvcs[i]
+		var nodeName string
+		if _, ok := pvc.Annotations[volume.AnnSelectedNode]; !ok {
+			return
+		}
+		nodeName = pvc.Annotations[volume.AnnSelectedNode]
+		var statefulSetReference metav1.OwnerReference
+		var podReference metav1.OwnerReference
+		for _, reference := range pvc.OwnerReferences {
+			if reference.Kind == "StatefulSet" {
+				statefulSetReference = reference
+			}
+			if reference.Kind == "Pod" {
+				podReference = reference
+			}
+		}
+		node, err := pl.sharedLister.NodeInfos().Get(nodeName)
```

Comment:
I think this line below can be placed in `filter`? We can store the reference message in state, since this logic is to handle the nodeInfo, I think place them in filter can reduce some delay between prefilter and filter. WDYT?

Also can we have some tests for this?

Reply:
`calPreFilterState` calculate match for each topology pair depend on nodes、pods、constraints,  i think should place in `prefilter`.  i add unit test when you feel ready to lgtm.